upstream.go

```diff
@@ -725,12 +756,19 @@ func (uc *upstreamConn) handleMessage(msg *irc.Message) error {
 			return fmt.Errorf("expected a prefix")
 		}
 
+		var reason string
+		if len(msg.Params) > 0 {
+			reason = msg.Params[0]
+		}
+
 		if msg.Prefix.Name == uc.nick {
 			uc.logger.Printf("quit")
 		}
 
 		for _, ch := range uc.channels {
 			delete(ch.Members, msg.Prefix.Name)
+
+			uc.AppendLog(ch.Name, "*** Quits: %s (%s@%s) (%s)", msg.Prefix.Name, msg.Prefix.User, msg.Prefix.Host, reason)
```

Comment:
We need to check whether the user belongs to the channel, otherwise we'll append this to all channel logs.